include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -1118,4 +1118,38 @@ def ZOp : OneTargetOp<"z", [Hermitian]> {
   }];
 }
 
+def UnitaryOp : QuakeOp<"unitary", [AttrSizedOperandSegments]> {
+  let summary = "General unitary operation specified by constant unitary matrix data";
+  let description = [{
+    This operation affects a general user-specified unitary operation, where 
+    the unitary can be specified at compile-time or at runtime via its matrix representation.
+
+    The matrix data must be provided as a row-major 1D array. It can be a 
+    compile-time known attribute, or a runtime value. 
+
+    If the unitary is provided at compile time, all target operands must 
+    have constant size (e.g. veq<?> will result in an error).
+
+    For compile-time unitary data, we represent as an ArrayAttr of DenseF32ArrayAttr 
+    elements, each element is a 2-tuple, the real and imaginary parts.
+  }];
+
+  let arguments = (ins StrAttr:$opName,              
+                       Variadic<AnyQType>:$controls, 
+                       Variadic<AnyQTargetType>:$targets, 
+                       OptionalAttr<ArrayAttr>:$constantUnitary,
+                       Optional<StdvecOf<[Complex<AnyFloat>]>>:$unitary,
+                       OptionalAttr<UnitAttr>:$is_adj,
+                       OptionalAttr<DenseBoolArrayAttr>:$negated_qubit_controls
+                       );
+  let results = (outs);
+
+  let assemblyFormat = [{
+    ( `<` `adj` $is_adj^ `>` )? (`[` $controls^ (`neg` $negated_qubit_controls^ )? `]`)?
+    $targets (`(` $unitary^ `)`)? `:` functional-type(operands, results) attr-dict
```

Comment:
```suggestion
    $targets (`,` $unitary^ )? `:` functional-type(operands, results) attr-dict
```
Remove the parens to keep the syntax looking a bit more consistent.

How is the constant unitary supposed to appear here?